src/main/java/kr/codesqaud/cafe/user/service/UserService.java

```diff
@@ -53,24 +53,19 @@ public List<UserResponse> getUserList() {
                 .collect(Collectors.toList());
     }
 
-    public void validateAndUpdateUser(UserUpdateForm userUpdateForm) {
-        if (checkPassword(userUpdateForm)) {
-            if (checkDuplicateName(userUpdateForm.getUserName())) {
-                userRepository.update(userUpdateForm.toEntity());
-            } else {
-                throw new UserUpdateFailedException("중복된 이름이 존재합니다.", "name");
-            }
-        } else {
-            throw new UserUpdateFailedException("비밀번호가 틀렸습니다.", "password");
-        }
+    public void updateUser(UserUpdateForm userUpdateForm) {
+        validateUpdateForm(userUpdateForm);
+        userRepository.update(userUpdateForm.toEntity());
     }
 
-    private boolean checkPassword(UserUpdateForm userUpdateForm) {
+    private void validateUpdateForm(UserUpdateForm userUpdateForm) {
         String password = userRepository.findByUserId(userUpdateForm.getUserId()).getPassword();
-        return userUpdateForm.getPassword().equals(password);
+        if (!userUpdateForm.getPassword().equals(password)) {
```

Comment:
password를 가져오지 않고 userUpdateForm에서 확인 하면 가독성이 조금 더 좋을 수 있다고 생각해요.
```suggestion
        if (!userUpdateForm.isSamePassword(password)) {
```